mountpoint-s3/src/prefetch/part_queue.rs

```diff
@@ -62,12 +68,18 @@ impl<E: std::error::Error + Send + Sync> PartQueue<E> {
         let part = if let Some(current_part) = current_part.take() {
             Ok(current_part)
         } else {
-            // Do `try_recv` first so we can track whether the read is starved or not
-            if let Ok(part) = self.receiver.try_recv() {
+            // Read from the auxiliary queue first if it's not empty
+            if !self.front_queue.is_empty() {
+                Ok(self
+                    .front_queue
+                    .pop_front()
+                    .expect("checked above that the queue is not empty"))
```

Comment:
```suggestion
            if let Some(part) = self.front_queue.pop_front() {
                Ok(part)
```

would get rid of `.expect` call I think